### DISCUSSION TITLE
Como aplicar semântica e acessibilidade ao projeto?

### DISCUSSION THREAD

**Discussion by napalmdev** [frontendbr/forum#22] [Discussão]
Fala galera, por vezes eu leio artigos relacionados a WAI-ARIA, Microdata e afins, e recentemente eu li este ótimo [artigo](http://blog.popupdesign.com.br/schema-org-dando-significado-ao-conteudo-na-internet/) da Dani Guerrato sobre o schema.org e achei muito bacana, porém como nunca participei de algum projeto de grande porte, nunca utilizei nada do tipo, mas agora acredito que terei que utilizar essas soluções, ai fica a dúvida: 

Como vocês aplicam essas soluções de semântica e acessibilidade em seus projetos? Vocês planejam e pesquisam sobre quais "items" irão utilizar em suas tags antes de começar a codar? Ou vão fazendo isso de forma linear conforme vão preparando suas páginas? Ou ainda se existe alguma forma automatizada de aplicá-las?

Enfim, não sei ficou meio confuso, mas é isso ai.

**Comment by felipefialho** [comment]
Ótimo!

Vamos lá. 

### Semântica

Acho que nem precisa aprofundar muito nessa questão né? O HTML deve ser estruturado para poder ser lido independente do CSS e do JavaScript. Os headings devem ser usados de forma lógica, assim como os elementos do HTML5. 

Mas também deve se tomar cuidado com relação ao uso exagerado desses novos elementos. Eles servem para dar semântica. Substituir `<div>`por `<section>` sem nenhum motivo só vai atrapalhar.

### Roles

Acho que [role](https://www.w3.org/wiki/PF/XTech/HTML5/RoleAttribute) é o minimo esperado. Devem fazer parte do desenvolvimento, logo em seguida da criação de um componente as roles devem ser adicionadas, de forma natural e automática. Isso já ajuda muito.

Obvio que existem milhares de roles, não é necessário adicionar todas. Mas pelo menos as macros.

### schema.org

Tenho usado muito o [schema.org](http://schema.org/), ajuda a dar ainda mais contexto ao projeto. Nesse caso ao adiciono no fim de tudo, pouco antes do projeto ser lançado, assim tenho uma ideia melhor do funcionamento de cada sessão.

### Navegação

Ponto importante, nós podemos adicionar :focus nas coisas para garantir que a navegação de teclado funcione bem, mas nem sempre é assim. Se você usa submenus ou qualquer situação de navegação que não pode ser controlada pelo :focus, é muito necessário fazer um fallback para o teclado usando JavaScript, assim garantimos acesso total ao conteúdo.

### Outros

Importante usar também atributos como aria-hidden, e etc, para ajudar o leitor de tela ou os mecanismos de busca a entenderem melhor o que pode ser descartado.

---

Fazer um site com o minimo de acessibilidade (pelo menos o que está do seu alcance), ajuda o próprio projeto a ser melhor indexado e ter uma estrutura melhor desenvolvida. Vale a pena ;)

**Comment by rickbenetti** [comment]
Sobre schema.org eu utilizo em todos os sites que faço e principalmente utilizei nas lojas virtuais que fiz [GrandCru](http://www.grandcru.com.br) foi um case legal, pois na época 2014 ninguém pensava em usar num ecommerce magento, a receita na maioria das empresas de magento é pegar o padrão e alterar pouca coisa eu tentei customizar ao máximo e fiz o possível pra aplicar semântica no código dando sentido do início ao fim.

Também para auxiliar desenvolvedores de lojas virtuais com magento (serve pra WooCommerce também) fiz um posts há um bom tempo exemplificando a utilização.

[Fique no topo da busca orgânica](http://rkb81.com.br/fique-no-topo-da-busca-organica/)

De mais a explicação do @LFeh ficou demais, não preciso acrescentar muito.

**Comment by dionarodrigues** [comment]
Ótima explicação @LFeh . Pensamos bem parecidos nessa questão.
E, @napalmdev , tendo em mente o que o @LFeh comentou, é sempre interessante planejar antes do desenvolvimento para que tudo funcione corretamente e evite ter que refazer trechos do HTML/JS.

**Comment by afonsopacifer** [comment]
Lembrando que quando falamos em acessibilidade, além do padrão WAI-ARIA temos também que dar uma olhada nas guidelines da W3C, para isso temos as  [Diretrizes de Acessibilidade para Conteúdo Web ou WCAG 2.0](https://www.w3.org/Translations/WCAG20-pt-br/).

Falando nisso, existe  um excelente projeto com vários links e dicas sobre o assunto: [awesome-a11y](https://github.com/brunopulis/awesome-a11y) :)

**Comment by napalmdev** [comment]
boa galera, +1 pros links @afonsopacifer , @diogorodrigues  vlw pelo toque, acho q vou acabar fazendo assim mesmo, @rkb81 legal cara, vou dar uma olhada nesse post ai. E @LFeh, sensacional cara resposta completíssima, vou avaliar com carinho e dar uma aprofundada nas opções ;D

**Comment by dionarodrigues** [comment]
Ótima dica esse projeto heim @afonsopacifer . Valeu!!

**Comment by felipefialho** [comment]
Muito bom @afonsopacifer!

**Comment by afonsopacifer** [comment]
Agradeçam a nosso amigo @brunopulis por ter criado essa referência irada :)

**Comment by brunopulis** [comment]
@afonsopacifer Opa valeu por divulgar o projeto! É super bem vindo os PR neles. E aproveitando também criei um [Slack](https://a11y-brazil.herokuapp.com/) sobre acessibilidade web quem quiser é só entrar, 

@napalmdev Quanto ao uso dos Roles da WAI-ARIA, eu recomendo dar uma olhada no blog do Paciello Group existem diversos artigos entre eles dois que você deve olhar.
- [Easy Content Organisation with HTLM5](https://www.paciellogroup.com/blog/2015/09/easy-content-organisation-with-html5/)
- [Anybody can be an a11y slacker](https://www.paciellogroup.com/blog/2015/07/anybody-can-be-an-a11y-slacker/)

O uso das roles deve ser feita com cuidado, atualmente não é mais necessário utilizar em blocos gerais como no header,  footer e etc, segundo o Steve Faulkner somente usar os roles em casos especificos, porque os browser's modernos colocam eles automaticamente. Vou procurar o artigo que ele fala isso e posto por aqui :)

**Comment by napalmdev** [comment]
Boa @brunopulis, vlw mesmo, vou dar uma olhada nesses artigos ai ______
